### PR TITLE
Development Add skeleton 

### PR DESCRIPTION
**Pull request by moriyad12**
### Description
Add project skeleton vue3

### Changes
adding components folder
adding utilities folder
adding api folder 
adding types folder

### Related Issues
https://github.com/codescalersinternships/test-tracker/issues/18

### CODE REVIEW

File: client/src/api/axios.ts
Comment (maayarosama) [inline client/src/api/axios.ts:1]: need to add another one instance with headers

File: client/src/components/test.ts
Comment (maayarosama) [inline client/src/components/test.ts:1]: components directory shouldn't contains ts files, only vue files

**Review by maayarosama — CHANGES_REQUESTED** [review]

File: client/src/api/axios.ts
Comment (moriyad12) [reply client/src/api/axios.ts:1]: done

File: client/src/components/test.ts
Comment (moriyad12) [reply client/src/components/test.ts:1]: done

File: client/src/api/axios.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import axios, { AxiosInstance } from 'axios';
+
+const axiosInstanceWithAuth: AxiosInstance = axios.create({
+  baseURL: 'https://your-api-base-url.com', 
+  timeout: 1000,
+  headers: {
+    'Content-Type': 'application/json',
+    'Authorization': 'Bearer ',
```
Comment (maayarosama) [inline client/src/api/axios.ts:8]:
```suggestion
      Authorization: "Bearer " + localStorage.getItem("token"),

```

File: client/src/api/axios.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import axios, { AxiosInstance } from 'axios';
+
+const axiosInstanceWithAuth: AxiosInstance = axios.create({
+  baseURL: 'https://your-api-base-url.com',
```
Comment (maayarosama) [inline client/src/api/axios.ts:4]:
```suggestion
  baseURL: import.meta.env.vite_app_endpoint, 

```

File: client/src/api/axios.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  headers: {
+    'Content-Type': 'application/json',
+    'Authorization': 'Bearer ', 
+  },
+});
+
+const axiosInstanceWithoutAuth: AxiosInstance = axios.create({
+  baseURL: 'https://your-api-base-url.com',
+  timeout: 1000,
+  headers: {
```
Comment (maayarosama) [inline client/src/api/axios.ts:15]: no need for this

File: client/src/api/axios.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  baseURL: 'https://your-api-base-url.com', 
+  timeout: 1000,
+  headers: {
+    'Content-Type': 'application/json',
+    'Authorization': 'Bearer ', 
+  },
+});
+
+const axiosInstanceWithoutAuth: AxiosInstance = axios.create({
+  baseURL: 'https://your-api-base-url.com',
```
Comment (maayarosama) [inline client/src/api/axios.ts:13]: same as above

File: client/src/api/axios.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+const axiosInstanceWithAuth: AxiosInstance = axios.create({
+  baseURL: 'https://your-api-base-url.com', 
+  timeout: 1000,
+  headers: {
+    'Content-Type': 'application/json',
+    'Authorization': 'Bearer ', 
+  },
+});
+
+const axiosInstanceWithoutAuth: AxiosInstance = axios.create({
```
Comment (maayarosama) [inline client/src/api/axios.ts:12]:
```suggestion
const BaseClient: AxiosInstance = axios.create({

```

File: client/src/api/axios.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import axios, { AxiosInstance } from 'axios';
+
+const axiosInstanceWithAuth: AxiosInstance = axios.create({
```
Comment (maayarosama) [inline client/src/api/axios.ts:3]:
```suggestion
const AuthClient: AxiosInstance = axios.create({

```

**Review by maayarosama — CHANGES_REQUESTED** [review]

**Review by maayarosama — APPROVED** [review]

**Pull request merged**